ui/src/test/java/edu/wpi/grip/ui/codegeneration/FindLinesGenerationTest.java

```diff
@@ -0,0 +1,107 @@
+package edu.wpi.grip.ui.codegeneration;
+
+import org.bytedeco.javacpp.opencv_core;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.opencv.core.Mat;
+import org.opencv.core.MatOfPoint;
+import org.opencv.core.Scalar;
+import org.opencv.imgproc.Imgproc;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import edu.wpi.grip.core.ManualPipelineRunner;
+import edu.wpi.grip.core.OperationMetaData;
+import edu.wpi.grip.core.Step;
+import edu.wpi.grip.core.operations.composite.FindLinesOperation;
+import edu.wpi.grip.core.operations.composite.HSLThresholdOperation;
+import edu.wpi.grip.core.operations.composite.LinesReport;
+import edu.wpi.grip.core.sockets.InputSocket;
+import edu.wpi.grip.core.sockets.OutputSocket;
+import edu.wpi.grip.core.sources.ImageFileSource;
+import edu.wpi.grip.ui.codegeneration.tools.HelperTools;
+import edu.wpi.grip.ui.codegeneration.tools.PipelineInterfacer;
+import edu.wpi.grip.util.Files;
+import grip_msgs.Lines;
+
+import static org.junit.Assert.assertTrue;
+
+@Category(GenerationTest.class)
+public class FindLinesGenerationTest extends AbstractGenerationTest {
+  private List<Number> hVal = new ArrayList<Number>();
+  private List<Number> sVal = new ArrayList<Number>();
+  private List<Number> lVal = new ArrayList<Number>();
+
+  public FindLinesGenerationTest() {
+    hVal.add(new Double(1.2));
+    hVal.add(new Double(51.0));
+    sVal.add(new Double(2.2));
+    sVal.add(new Double(83.2));
+    lVal.add(new Double(1.0));
+    lVal.add(new Double(101.0));
+  }
+
+  void generatePipeline() {
+    Step step0 = gen.addStep(new OperationMetaData(HSLThresholdOperation.DESCRIPTION, () -> new
+        HSLThresholdOperation(isf, osf)));
+    ImageFileSource img = loadImage(Files.imageFile);
+    OutputSocket imgOut0 = pipeline.getSources().get(0).getOutputSockets().get(0);
+
+    for (InputSocket sock : step0.getInputSockets()) {
+      if (sock.getSocketHint().isCompatibleWith(imgOut0.getSocketHint())) {
+        gen.connect(imgOut0, sock);
+      } else if (sock.getSocketHint().getIdentifier().equals("Hue")) {
+        sock.setValue(hVal);
+      } else if (sock.getSocketHint().getIdentifier().equals("Saturation")) {
+        sock.setValue(sVal);
+      } else if (sock.getSocketHint().getIdentifier().equals("Luminance")) {
+        sock.setValue(lVal);
+      }
+    }
+
+    Step step1 = gen.addStep(new OperationMetaData(FindLinesOperation.DESCRIPTION, () -> new
+        FindLinesOperation(isf, osf)));
+    OutputSocket imgOut1 = pipeline.getSteps().get(0).getOutputSockets().get(0);
+    for (InputSocket sock : step1.getInputSockets()) {
+      if (sock.getSocketHint().isCompatibleWith(imgOut1.getSocketHint())) {
+        gen.connect(imgOut1, sock);
+      }
+    }
+  }
+
+/*  @Test
```

Comment:
Don't comment out tests, just annotate them with `@Ignore("[reason why it's ignored]")`. Otherwise diffs blow up and it's difficult to tell if something got changed